shared/src/main/scala/mlscript/Typer.scala

```diff
@@ -476,6 +476,8 @@ class Typer(var dbg: Boolean, var verbose: Bool, var explainErrors: Bool) extend
         decls.foreach { case nme -> mt => 
           implicit val prov: TypeProvenance = mt.prov
           td.baseClasses.foreach { case Var(bn) => ctx.mthDecls.get(bn).foreach(_.get(nme).foreach(ss(mt, _, S(bn)))) }
+          if (!defs.isDefinedAt(nme))
+            ctx.env -= tn.name
```

Comment:
It is pretty hacky to remove the ctor from the context after the fact, instead of not registering it in the first place. In fact, it leads to false positives, as this should be rejected:

```ts
class Test7A
    method Mth7A: nothing
class Test7B
    method Mth7B = Test7A
```

but is currently accepted!